feat: rework trigger workflow docs (#764)
knocklabs/docs: content/send-notifications/triggering-workflows/events.mdx

---
title: Triggering workflows via an event
description: Learn more about how to trigger cross-channel notification workflows in Knock via an event.
tags: ["trigger", "source", "event", "segment", "cdp", "customer data platform"]
section: Send notifications
---

> Workflows can be triggered by a stream of incoming events, from a customer data platform like [Segment](/integrations/sources/segment), or from a direct HTTP webhook integration.

Reviewer: ```suggestion
Workflows can be triggered by a stream of incoming events from a customer data platform like [Segment](/integrations/sources/segment), or from a direct HTTP webhook integration.
```

> Triggering workflows via events gives you a powerful way to decouple your notification workflows from your backend systems, allowing non-developers in the organization the ability to setup and configure notifications driven by the business events happening in your product or service.

Reviewer: ```suggestion
Triggering workflows via events gives you a powerful way to decouple your notification workflows from your backend systems, allowing non-developers in the organization the ability to set up and configure notifications driven by the business events happening in your product or service.
```

Event triggered workflows require that you have one or more [Sources](/integrations/sources/overview) configured and connected to Knock.

## Configuring an event trigger

> You can create and manage event triggers for your workflows in the "Sources" section (under "Developers" in the sidebar) or directly from the workflow builder when you click the "Trigger" step.

Reviewer 2: ```suggestion
You can create and manage event triggers for your workflows in the **Sources** section (under **Developers** in the sidebar) or directly from the workflow builder when you click the "Trigger" step.
```

> You can create and manage event triggers for your workflows in the "Sources" section (under "Developers" in the sidebar) or directly from the workflow builder when you click the "Trigger" step.

Reviewer 3: ```suggestion
You can create and manage event triggers for your workflows in the "Sources" section (under "Developers" in the sidebar) of your dashboard, or directly from the workflow builder when you click the "Trigger" step.
```

> From the Trigger step sidebar, if you have events connected to Knock you'll see the option to switch the trigger type to "Event" from the dropdown menu. Once here, you can select an event that when received will trigger this workflow. You can also map the critical fields needed to run a workflow to the fields that will be in the incoming event payload.

Reviewer: ```suggestion
From the Trigger step sidebar, if you have events connected to Knock you'll see the option to switch the trigger type to "Event" from the dropdown menu. Once here, you can select an event that will trigger this workflow when it's received. You can also map the critical fields needed to run a workflow to the fields that will be in the incoming event payload.
```

[Learn more about sources](/integrations/sources/overview)

## Frequently asked questions

<AccordionGroup>
  <Accordion title="Can I trigger a workflow for multiple recipients from a single event?">
    Yes, it's possible to override the default behavior of a source event
    trigger to point the trigger to a property on the event that resolves to a
    list of recipients.
  </Accordion>
  <Accordion title="Can I cancel a workflow run triggered by an event?">
    No, it's not currently possible to cancel a workflow run triggered by an
    event. You must still use the `cancel` endpoint to cancel a workflow run.

>   </Accordion>

Reviewer: This is a little confusing. Are you saying you can't use another event to cancel a current workflow?

Reviewer 2: Agree with Meryl
